lib/TH/THTensor.c

```diff
@@ -1,6 +1,11 @@
 #include "THAtomic.h"
 #include "THTensor.h"
+
+#include "THVectorSIMD.c"
```

Comment:
I believe these includes have nothing to do with tensor logic and are better off in `THVectorDispatch.c`. Maybe it'd be better to rename it to `THVector.c` and compile both the dispatch logic and implementations together.